src/structs/lepton_header.rs

```diff
@@ -95,77 +102,81 @@ impl LeptonHeader {
             max_bpos: 0,
             max_sah: 0,
             jpeg_file_size: 0,
-            plain_text_size: 0,
-            uncompressed_lepton_header_size: 0,
+            uncompressed_lepton_header_size: None,
+            git_revision_prefix: [0u8; 4],
+            encoder_version: 0,
         };
     }
 
-    /// reads the start of the lepton file and parses the compressed header. Returns the raw JPEG header contents.
-    pub fn read_lepton_header<R: Read>(
+    pub fn read_lepton_fixed_header(
         &mut self,
-        reader: &mut R,
+        header: &[u8; FIXED_HEADER_SIZE],
         enabled_features: &mut EnabledFeatures,
-    ) -> Result<()> {
-        let mut header = [0 as u8; LEPTON_FILE_HEADER.len()];
-
-        reader.read_exact(&mut header).context(here!())?;
-
-        if !buffer_prefix_matches_marker(header, LEPTON_FILE_HEADER) {
+    ) -> Result<usize> {
+        if header[0..2] != LEPTON_FILE_HEADER[0..2] {
             return err_exit_code(ExitCode::BadLeptonFile, "header doesn't match");
         }
-
-        // Complicated logic of version compatibility should be verified by the caller.
-        // Currently just matching the version version.
-        let version = reader.read_u8().context(here!())?;
-        if version != LEPTON_VERSION {
+        if header[2] != LEPTON_VERSION {
             return err_exit_code(
                 ExitCode::VersionUnsupported,
-                format!("incompatible file with version {0}", version).as_str(),
+                format!("incompatible file with version {0}", header[3]).as_str(),
             );
         }
-
-        let mut header = [0 as u8; 21];
-        reader.read_exact(&mut header).context(here!())?;
-
-        // Z = baseline non-progressive
-        // Y = chunked encoding of a slice of a JPEG (not supported yet)
-        // X = progressive
-        if header[0] != LEPTON_HEADER_BASELINE_JPEG_TYPE[0]
-            && header[0] != LEPTON_HEADER_PROGRESSIVE_JPEG_TYPE[0]
+        if header[3] != LEPTON_HEADER_BASELINE_JPEG_TYPE[0]
+            && header[3] != LEPTON_HEADER_PROGRESSIVE_JPEG_TYPE[0]
         {
             return err_exit_code(
                 ExitCode::BadLeptonFile,
-                format!("Unknown filetype in header {0}", header[0]).as_str(),
+                format!("Unknown filetype in header {0}", header[4]).as_str(),
             );
         }
 
-        let mut c = Cursor::new(header);
+        // header[4] is the number of threads, but we don't care about that
```

Comment:
Hmm, it is the number of separate streams in the Lepton file, not of threads. However, it contains garbage in legacy files, and is duplicated by HH header in non-legacy files, so is not needed - only if you want to make a sanity crosscheck with HH header value.